### PR TITLE
DDPB-2244 report status db-cached

### PR DESCRIPTION
**Pull request by elvisciotti**

### CODE REVIEW

File: src/AppBundle/Controller/Report/AccountController.php [hunk 1]
```diff
@@ -31,7 +31,10 @@ public function addAccountAction(Request $request, $reportId)
 
         $this->fillAccountData($account, $data);
 
+        $report->updateSectionStatus(Report::SECTION_BANK_ACCOUNTS);
```
Comment (slizzio) [inline src/AppBundle/Controller/Report/AccountController.php]: Optional: have a sectionId as a private property in each controller:
private $sectionId = Report::SECTION_BANK_ACCOUNTS;
then your update becomes:
$report->updateSectionStatus($this->sectionId);

File: src/AppBundle/Entity/Report/Report.php [hunk 2]
```diff
@@ -393,16 +394,6 @@ public static function getSectionsSettings()
...
 
-    /**
-     * set Due date to +8 weeks after end date
-     */
-    public function updateDueDateBasedOnEndDate()
-    {
-        // due date set to 8 exactly weeks (56 days) after the start date
-        $this->dueDate = clone $this->endDate;
-        $this->dueDate->add(new \DateInterval('P56D'));
-    }
```
Comment (slizzio) [inline src/AppBundle/Entity/Report/Report.php]: why remove this?

File: src/AppBundle/Entity/Report/Traits/StatusTrait.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    /**
+     * @JMS\VirtualProperty
+     * @JMS\Groups({
+     *     "status",
+     *     "report-status",
+     *     "decision-status",
+     *     "contact-status",
+     *     "visits-care-state",
```
Comment (slizzio) [inline src/AppBundle/Entity/Report/Traits/StatusTrait.php]: should be either status or state but not both. Confusing. Id go with "status"

File: src/AppBundle/Service/ReportStatusService.php [hunk 4]
```diff
@@ -5,14 +5,20 @@
...
+ */
 class ReportStatusService
 {
     const STATE_NOT_STARTED = 'not-started';
     const STATE_INCOMPLETE = 'incomplete';
     const STATE_DONE = 'done';
     const STATE_NOT_MATCHING = 'not-matching'; //only used for balance section
     const STATE_EXPLAINED = 'explained'; //only used for balance section
-
+    const ENABLE_SECTION_STATUS_DB_CACHE = true;
```
Comment (slizzio) [inline src/AppBundle/Service/ReportStatusService.php]: if this is a feature flag it should be named the same as the feature: ie ENABLE_STATUS_CACHE

File: src/AppBundle/Service/ReportStatusService.php [hunk 5]
```diff
@@ -470,51 +489,57 @@ public function getRemainingSections()
         }) ?: [];
     }
```
Comment (slizzio) [inline src/AppBundle/Service/ReportStatusService.php]: can this be removed entirely?

File: tests/AppBundle/Controller-Report/AccountControllerTest.php [hunk 6]
```diff
@@ -162,14 +165,16 @@ public function testEdit()
...
+        $account = self::fixtures()->clear()->getRepo('Report\BankAccount')->find(self::$account1->getId());
         $this->assertEquals('bank1-modified', $account->getBank());
         $this->assertTrue($account->getIsClosed());
         $this->assertEquals('yes', $account->getIsJointAccount());
 
         // assert user cannot modify another users' account
         $url2 = '/account/' . self::$account2->getId();
         $this->assertEndpointNotAllowedFor('PUT', $url2, self::$tokenDeputy);
+
+        $this->assertArrayHasKey('state', self::fixtures()->getReportFreshSectionStatus(self::$report1, Report::SECTION_BANK_ACCOUNTS));
```
Comment (slizzio) [inline tests/AppBundle/Controller-Report/AccountControllerTest.php]: can we assert the status is correct? Not just that it has one.

File: src/AppBundle/Controller/Report/AccountController.php (hunk 1 above)
Comment (elvisciotti) [reply src/AppBundle/Controller/Report/AccountController.php]: that was done

File: src/AppBundle/Entity/Report/Traits/StatusTrait.php (hunk 3 above)
Comment (elvisciotti) [reply src/AppBundle/Entity/Report/Traits/StatusTrait.php]: it was all named wrong since the start :)
I'd postpone refactoring for priority reasons

**Pull request merged**